src/Finder.elm

```diff
@@ -63,25 +64,25 @@ import UI.Modal as Modal
 
 type FinderSearch
     = NotAsked
-    | Searching (Maybe FinderSearchResults)
-    | Success FinderSearchResults
-    | Failure Http.Error
+    | Searching String (Maybe FinderSearchResults)
+    | Success String FinderSearchResults
+    | Failure String Http.Error
```

Comment:
Tracking the string means we don't update the empty state and error state screens while typing a new query - the error state should show the query it errored for, which might differ from the most recent input.